raft-tech/TANF-app
# DIGIT Team Group + Kibana + Queries

## Summary of Changes
- Added DIGIT Team group and gave datafile, datafile summary, and parser error view only permissions
- Allowed DIGIT Team group to access Kibana
- Provide MD with caveats of Kibana data export, along with OFA queries that have been converted to KQL and EQL
Pull request closes #1441 

## How to Test

- Uncomment `REACT_APP_DEV_KIBANA` from the frontend `.env` file
- Set `BYPASS_KIBANA_AUTH=True` in the backend `.env`

```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

1. Request access via the frontend with non-superuser account
2. Log in as superuser and set the non-superusers group as `DIGIT Team` check `is_staff`
3. Log back in as non-superuser
4. Verify DIGIT user can't submit datafiles, and can view only the aforementioned objects in the DAC
5. Verify the DIGIT user can navigate to Kibana
6. As a user who can submit data, load the DB up with some records
7. Verify the example queries (potentially with modifications depending on what datafile you submit and as what STT) execute and produce the desired result.

[OFA_to_Kibana_Queries_with_Export.md](https://github.com/raft-tech/TANF-app/files/14379218/OFA_to_Kibana_Queries_with_Export.md)


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] User group should be able to access all raw data files in S3  via: `/admin/data_files/*`
- [x] User group should be able to access the most recent version of parsed and validated data that has been accepted into the ~~Elastisearch~~ postgres db 
- [x] User group can access Kibana
- [x] User group can execute converted OFA SQL queries


### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
